lib/bookingsync/engine/auth_helpers.rb

```diff
@@ -11,7 +11,13 @@ module BookingSync::Engine::AuthHelpers
 
   # @return [Account, nil] currently authorized Account or nil if unauthorized
   def current_account
-    @current_account ||= ::Account.find_by(synced_id: session[:account_id]) if session[:account_id].present?
+    if session[:account_id].present?
+      if BookingSync::Engine.support_multi_applications?
+        @current_account ||= ::Account.find_by(host: request.host, synced_id: session[:account_id])
```

Comment:
how about having a method like `find_by_host_and_account_id`?

Reply:
What's wrong with this way?

Reply:
Ideally, ActiveRecord query interface would stay private. When using all these finders outside AR models, you basically couple every single part of the application to the database. AuthHelpers doesn't sound like something that should have a dependency on the database, so AR query method shouldn't be used here. We should just use domain methods, repository-patterns style. And that way, we would be decoupled from DB because under the hood, there might be e.g. in-memory adapter instead of the DB one.